### PR TITLE
Fix bulk ingest NPE with empty pipeline 

### PR DESCRIPTION
**Pull request by chishui**
### Description
Fix NPE when bulk ingest with empty pipeline

### Related Issues
#15032

### Check List
- [x] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/ingest/IngestService.java [hunk 1]
```diff
@@ -1271,6 +1271,14 @@ private static IngestDocumentWrapper toIngestDocumentWrapper(int slot, IndexRequ
         return new IngestDocumentWrapper(slot, toIngestDocument(indexRequest), null);
     }
 
+    private static List<IngestDocumentWrapper> toIngestDocumentWrappers(List<Integer> slots, List<IndexRequest> indexRequests) {
```
Comment (mch2) [inline server/src/main/java/org/opensearch/ingest/IngestService.java:1274]: I don't think this needs to be static?

File: server/src/main/java/org/opensearch/ingest/IngestService.java [hunk 2]
```diff
@@ -1271,6 +1271,14 @@ private static IngestDocumentWrapper toIngestDocumentWrapper(int slot, IndexRequ
...
+        List<IngestDocumentWrapper> ingestDocumentWrappers = new ArrayList<>();
+        for (int i = 0; i < slots.size(); ++i) {
+            ingestDocumentWrappers.add(toIngestDocumentWrapper(slots.get(i), indexRequests.get(i)));
+        }
+        return ingestDocumentWrappers;
```
Comment (mch2) [inline server/src/main/java/org/opensearch/ingest/IngestService.java:1279]: nitpick, return immutable list instead:
```java
        return slots.stream()
            .map(i -> toIngestDocumentWrapper(i, indexRequests.get(i)))
            .collect(Collectors.toUnmodifiableList());
```

File: server/src/test/java/org/opensearch/ingest/IngestServiceTests.java [hunk 3]
```diff
@@ -1995,6 +1995,43 @@ public void testExecuteBulkRequestInBatchWithDefaultBatchSize() {
...
+        IndexRequest indexRequest2 = new IndexRequest("_index").id("_id2").source(emptyMap()).setPipeline("_id").setFinalPipeline("_none");
+        bulkRequest.add(indexRequest2);
+        IndexRequest indexRequest3 = new IndexRequest("_index").id("_id3").source(emptyMap()).setPipeline("_none").setFinalPipeline("_id");
+        bulkRequest.add(indexRequest3);
+        IndexRequest indexRequest4 = new IndexRequest("_index").id("_id4").source(emptyMap()).setPipeline("_id").setFinalPipeline("_none");
+        bulkRequest.add(indexRequest4);
+        @SuppressWarnings("unchecked")
+        final BiConsumer<Integer, Exception> failureHandler = mock(BiConsumer.class);
+        @SuppressWarnings("unchecked")
+        final BiConsumer<Thread, Exception> completionHandler = mock(BiConsumer.class);
```
Comment (andrross) [inline server/src/test/java/org/opensearch/ingest/IngestServiceTests.java:2021]: No need to mock with unchecked casts. Make them maps and pass the `put` method reference, then assert one of them is empty and the other one has one entry with the current thread as the key.

File: server/src/main/java/org/opensearch/ingest/IngestService.java [hunk 4]
```diff
@@ -997,7 +997,7 @@ private void innerBatchExecute(
         Consumer<List<IngestDocumentWrapper>> handler
     ) {
         if (pipeline.getProcessors().isEmpty()) {
-            handler.accept(null);
+            handler.accept(toIngestDocumentWrappers(slots, indexRequests));
```
Comment (andrross) [inline server/src/main/java/org/opensearch/ingest/IngestService.java:1000]: Codecov is claiming this is not covered, which suggests your unit test isn't actually testing this case. Is that right?

File: server/src/main/java/org/opensearch/ingest/IngestService.java (hunk 1 above)
Comment (andrross) [reply server/src/main/java/org/opensearch/ingest/IngestService.java:1274]: Personally I have no problem making methods static if they don't operate on any instance state. Any reason to prefer a non-static method?

File: server/src/main/java/org/opensearch/ingest/IngestService.java (hunk 2 above)
Comment (chishui) [reply server/src/main/java/org/opensearch/ingest/IngestService.java:1279]: The `i` in `map` is the value in slots not the index which makes `indexRequests.get(i)` incorrect.

Comment (mch2) [reply server/src/main/java/org/opensearch/ingest/IngestService.java:1279]: ah my mistake, you could still make this immutable but feel free to ignore esp bc this is coming from an internal method.

**Comment by chishui** [comment]
> FAILURE

Failing due to - :distribution:bwc:staged:buildBwcLinuxTar - https://github.com/opensearch-project/OpenSearch/issues/7396

**Comment by chishui** [comment]
> ❌ Gradle check result for [9e21d1f](https://github.com/opensearch-project/OpenSearch/commit/9e21d1f31d2f6e191b4eb8b179308a3b37f0be1c): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/44025/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

#15117

**Comment by chishui** [comment]
@mch2 @andrross could you help review again?

**Review by andrross — APPROVED** [review]

**Pull request merged**